fuse_examples/imaging/oai_example/downstream/classification.py

```diff
@@ -0,0 +1,329 @@
+import sys
+import copy
+from fuse.dl.losses.loss_default import LossDefault
+from fuse.dl.lightning.pl_module import LightningModuleDefault
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import LearningRateMonitor
+import hydra
+from omegaconf import DictConfig
+from clearml import Task
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))  # noqa
+from data.oai_ds import OAI
+from fuse.dl.models.backbones.backbone_unet3d import UNet3D
+from sklearn.metrics import mean_absolute_error
+
+torch.set_float32_matmul_precision("medium")
+torch.use_deterministic_algorithms(False)
+
+
+@hydra.main(version_base="1.2", config_path=".", config_name="classification_config")
+def main(cfg: DictConfig):
+    cfg = hydra.utils.instantiate(cfg)
+    results_path = cfg.results_dir
+
+    model_dir = os.path.join(results_path, cfg.experiment)
+    if not os.path.isdir(model_dir):
+        os.makedirs(model_dir)
+
+    # set constant seed for reproducibility.
+    os.environ[
+        "CUBLAS_WORKSPACE_CONFIG"
+    ] = ":4096:8"  # required for pytorch deterministic mode
+    rand_gen = Seed.set_seed(1234, deterministic_mode=True)
+
+    cls_targets = cfg.cls_targets
+    tags = [cfg.task, cfg.backbone]
+    print(f"EXPERIMENT : {cfg.experiment}")
+    if cfg.clearml:
+        if len(glob(model_dir + "/*.cmlid")) == 0:
+            task = Task.create(
+                project_name=cfg.clearml_project_name, task_name=cfg.experiment
+            )
+            with open(os.path.join(model_dir, f"{task.id}.cmlid"), "w"):
+                pass
+
+        clearml_task_id = glob(model_dir + "/*.cmlid")[0].split("/")[-1].split(".")[0]
+        task = Task.init(
+            project_name=cfg.clearml_project_name,
+            task_name=cfg.experiment,
+            reuse_last_task_id=clearml_task_id,
+            continue_last_task=0,
+            tags=tags,
+        )
+        task.connect(cfg)
+
+    if cfg.backbone == "unet3d":
+        backbone = UNet3D(for_cls=True)
+
+        if os.path.exists(cfg.ckpt_path):
+            print(f"LOADING ckpt from {cfg.ckpt_path}")
+            state_dict = torch.load(
+                open(cfg.ckpt_path, "rb"), map_location=torch.device("cpu")
+            )
+            state_dict = {
+                k.replace("teacher_backbone.", ""): v
+                for k, v in state_dict["state_dict"].items()
+                if "teacher_backbone." in k
+            }
+        elif cfg.pretrained:
+            state_dict = torch.load(
+                cfg.pretrained_pth, map_location=torch.device("cpu")
+            )
+            state_dict = {
+                k.replace("module.backbone.", ""): v
+                for k, v in state_dict["net"].items()
+            }
+        else:
+            state_dict = backbone.state_dict()
+
+        backbone.load_state_dict(state_dict, strict=False)
+        conv_inputs = [("model.backbone_features", 512)]
+
+    ## FuseMedML dataset preparation
```

Comment:
Just nice to have: moving all the logic to DataModule.

Reply:
because each of the files has its own handling of the data for me it is easier when the unique lines is just separated on different files.